test/test_serre_green_naghdi_1d.jl

```diff
@@ -59,6 +149,96 @@ EXAMPLES_DIR = joinpath(examples_dir(), "serre_green_naghdi_1d")
 
         @test_allocations(semi, sol, allocs=450_000)
     end
+
+    @trixi_testset "serre_green_naghdi_soliton_relaxation.jl with bathymetry_mild_slope" begin
+        # same as serre_green_naghdi_soliton_relaxation.jl but with more allocations
+        @test_trixi_include(joinpath(EXAMPLES_DIR,
+                                     "serre_green_naghdi_soliton_relaxation.jl"),
+                            tspan=(0.0, 0.1),
+                            bathymetry_type=bathymetry_mild_slope,
+                            l2=[8.252225169608892e-8, 6.724994488577288e-7, 0.0],
+                            linf=[2.6716495016287922e-8, 9.642466235713909e-8, 0.0],
+                            cons_error=[2.842170943040401e-14, 4.649614027130156e-13, 0.0],
+                            change_waterheight=2.842170943040401e-14,
+                            change_entropy_modified=0.0)
+
+        @test_allocations(semi, sol, allocs=850_000)
+    end
+
+    @trixi_testset "serre_green_naghdi_soliton_relaxation.jl with bathymetry_variable" begin
+        # same as serre_green_naghdi_soliton_relaxation.jl but with more allocations
+        @test_trixi_include(joinpath(EXAMPLES_DIR,
+                                     "serre_green_naghdi_soliton_relaxation.jl"),
+                            tspan=(0.0, 0.1),
+                            bathymetry_type=bathymetry_variable,
+                            l2=[8.252225169608892e-8, 6.724994488577288e-7, 0.0],
+                            linf=[2.6716495016287922e-8, 9.642466235713909e-8, 0.0],
+                            cons_error=[2.842170943040401e-14, 4.649614027130156e-13, 0.0],
+                            change_waterheight=2.842170943040401e-14,
+                            change_entropy_modified=0.0)
+
+        @test_allocations(semi, sol, allocs=850_000)
+    end
+
+    @trixi_testset "serre_green_naghdi_well_balanced.jl" begin
+        @test_trixi_include(joinpath(EXAMPLES_DIR,
+                                     "serre_green_naghdi_well_balanced.jl"),
+                            tspan=(0.0, 2.0),
+                            l2=[0, 0, 0],
+                            linf=[0, 0, 0],
+                            cons_error=[0, 0, 0],
+                            change_waterheight=0.0,
+                            change_momentum=0.0,
+                            change_entropy_modified=0.0,
+                            lake_at_rest=0.0)
```

Comment:
Out of curiosity: In reality we get values around machine precision and not exactly 0, right?

Reply:
That's what I would expect in general but here we get

```julia
julia> trixi_include("examples/serre_green_naghdi_1d/serre_green_naghdi_well_balanced.jl");
[...]

────────────────────────────────────────────────────────────────────────────────────────────────────
Simulation running 'SerreGreenNaghdiEquations1D-BathymetryVariable' with 'initial_condition_discontinuous_well_balancedness'
────────────────────────────────────────────────────────────────────────────────────────────────────
 #timesteps:                 60                run time:       3.35273791e-01 s
 Δt:             5.00000000e-01                └── GC time:    1.25329125e-01 s (37.381%)
 sim. time:      3.00000000e+01 (100.000%)
 #DOF:                      512                alloc'd memory:        249.217 MiB

 Variable:       η                v                D             
 L2 error:       0.00000000e+00   0.00000000e+00   0.00000000e+00
 Linf error:     0.00000000e+00   0.00000000e+00   0.00000000e+00
 |∫q - ∫q₀|:     0.00000000e+00   0.00000000e+00   0.00000000e+00

 Integrals:    
 ∫η          :   4.00000000e+00
 ∫P          :   0.00000000e+00
 ∫U_modified :   4.00000000e+00
 ∫|η-η₀|     :   0.00000000e+00
[...]
```

Reply:
Ok, I remember that I had something similar once when I forgot to set $\eta_0\neq 0$, which means everything is just exactly zero all the time, but didn't really test true well-balancedness. This is not the case here, but do you have another idea why this is exactly zero?

Reply:
No further idea, sorry

Reply:
Ok, I guess it's just fine then. Having _too_ good results isn't too bad :sweat_smile: